_docs_agent/plugins/nodejs.md

```diff
@@ -4,46 +4,89 @@ title: Nodejs support
+app.use((req, res, next) => {
+  res.locals.startEpoch = Date.now()
+  next()
+})
 {% endhighlight %}
 
-Next we enable morgan and set a custom log format.
+The next code to add is to expose the Prometheus endpoint, it should be add just above the `app.listen` code line.
 
 {% highlight javascript %}
-morgan('nodejs ":method :url HTTP/:http-version" :response-time :status') 
+// Provide prometheus endpoint to collect data
+app.get('/metrics', (req, res) => {
+  // Deny connections from outside the container
+  var remote = req.ip || req.connection.remoteAddress
+  if ((remote !== '::1') && (remote !== 'localhost') && (remote !== '::ffff:127.0.0.1')) {
+    return res.status(401).send();
+  }
+
+  // Return prometheus metrics
+  res.set('Content-Type', promClient.register.contentType)
+  res.end(promClient.register.metrics())
+})
+
+// Register stop time for each request and pass to prometheus
+app.use((req, res, next) => {
+  const responseTimeInMs = Date.now() - res.locals.startEpoch
+
+  promLatency
+    .labels(req.method, req.route.path, res.statusCode)
+    .observe(responseTimeInMs)
+  promRate
+    .labels(req.method, req.route.path, res.statusCode)
+    .inc()
+
+  next()
+})
 {% endhighlight %}
 
-The log format has been adapted to work with the [CoScale Apache plugin]({{ site.baseurl }}/agent/plugins/apache). 
-
-### Configure the CoScale Apache plugin to parse the requests
-
-CoScale uses the Apache plugin to parse the information from the NodeJS request log, go to agent configuration and enable the Apache plugin for your environment. Depending on the setup you will need to add the plugin to an agent or setup the CoScale in-container monitoring for a certain image.
-
-#### Global statistics
-
-The NodeJS configuration we've setup does not support an active status page, so you can disable this functionality. 
+### Configure the CoScale Prometheus plugin to parse the information from the endpoint
 
-#### Log configuration
+CoScale uses the Prometheus plugin to parse the information from the NodeJS Prometheus endpoint, go to Agent configuration and enable the CoScale Prometheus plugin for your environment. Depending on the setup you will need to add the plugin to an agent or setup the CoScale in-container monitoring for a certain image.
 
-The default log format in the Apache plugin will work with the configuration that we enabled for NodeJS with morgan. So you just need to point the plugin to the right logfile on your filesystem or use `/dev/stdout` if you're using Docker logs.
+The endpoint configured above is available on the default nodejs port (80) with endpoint `metrics`:
 
-### Active checks
+`http://localhost:80/metrics`
 
-The Apache plugin can be configured to perform a HTTP GET on your NodeJS. This active monitoring allows us to calculate the uptime of the service and the response time of the provided URL.
+You can also add a Docker Label to your `Dockerfile` to automatically pick up the Prometheus endpoint:
```

Comment:
your `Dockerfile` so that the agent will automatically pick up the Prometheus endpoint when it detects a new container*